dipy/reconst/cti.py

```diff
@@ -0,0 +1,658 @@
+                                diffusion_tensor[0, 1],
+                                diffusion_tensor[0, 2],
+                                diffusion_tensor[1, 2]])
+
+        return dt_elements
+    
+    @property
+    def cvt(self):  # created
+        """
+        Returns the 21 independent elements of the covariance tensor as an array
+        """
+        return self.model_params[27:48, ...]
+
+   # def cvt(__):  # calculates the mean of all covariance parameters. required ? Formula ?
+
+        # There are 4 (radial,mean,axial,fractional) do we have something similar for covariance in qti ?
+
+        # we separate the kurtosis in 3 parts: isotropic+anisotropoic+microscopic. Do we need methods for this? : REFER VIDEO ON THIS
+
+    def predict(self, gtab1, gtab2, S0=100):  # created
+        """Given a CTI model fit, predict the signal on the vertices of a gradient table
+
+        Parameters:
+        -----------
+        params: numpy.ndarray (...,43)
+                All parameters estimated from the correlation tensor model.
+                Paramters are ordered as follows::
+
+                1. Three diffusion tensor's eigenvalues
+                2. Three lines of the eigenvector matrix each containing the
+                first, second and third coordinates of the eigenvector
+                3. Fifteen elements of the kurtosis tensor
+                4. Twenty-One elements of the covariance tensor
+        gtab1: dipy.core.gradients.GradientTable
+        A GradientTable class instance for first DDE diffusion epoch
+        gtab2: dipy.core.gradients.GradientTable
+        A GradientTable class instance for second DDE diffusion epoch
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+
+        Returns
+        -------
+        S : numpy.ndarray
+            Signals.
+        """
+        return cti_prediction(self.model_params, gtab1, gtab2, S0)
+        
+        #def calculate_K_aniso(D, C):  # D is a diffusion tensor 3x3, C is a 1D array.
+
+    @auto_attr
+    def K_aniso(self):
+        r""" Returns the anisotropic Source of Kurtosis (K_aniso) 
+            
+            Notes 
+            -----
+            The K_aniso is defined as : 
+            
+            :math:: 
+            
+            \[K_{aniso} = \frac{6}{5} \cdot \frac{\langle V_{\lambda}(D_c) \rangle}{\overline{D}^2}\]
+            
+        where: \(K_{aniso}\) is the anisotropic kurtosis, 
+            \(\langle V_{\lambda}(D_c) \rangle\) represents the mean of the variance of eigenvalues of the diffusion tensor,
+            \(\overline{D}\) is the mean of the diffusion tensor.
+        """ 
+
+        D = self.dft
+        C = self.cvt
+        matrix = np.array([[D[0], D[3], D[4]],[D[3], D[1], D[5]],[D[4], D[5], D[2]]])
+        D  = matrix 
+        
+        Variance = 2/9 * (C[0] + D[0, 0] ** 2 + C[1] + D[1, 1] ** 2 + C[2] + D[2, 2]**2 - C[5]
```

Comment:
this will only work for single voxel data. For multi voxel data you have to call the tensor elements in the following way:

C[..., 0]
D[..., 0] (assuming that tensor elements are in the last dimension according to self.lower_triangular).

...
(and so on)